src/Core/Addon/Theme/Theme.php

```diff
@@ -75,4 +86,20 @@ public function onReset()
     {
         return true;
     }
+
+    public function setPageLayouts(array $layouts)
+    {
+        $this->settings['page_layouts'] = $layouts;
+    }
+
+    public function getLayoutForPage($entity)
+    {
+        $layout_name = $this->theme_settings['default_layout'];
+        if ($this->settings['page_layouts'][$entity]) {
```

Comment:
It's likely we will be asked to get a layout for a page we didn't know about at some point, so `$this->settings['page_layouts'][$entity]` is probably not always set and will result in a notice.